### PR TITLE
Wrap calls to ga check for analytics_tracking_id

### PR DESCRIPTION
**Pull request by jason-jackson**
### Changed
- Wrap calls to ga check for analytics_tracking_id

### CODE REVIEW

**Review by briskt — APPROVED** [review]
Great job hunting down all the uses of `ga`!

**Comment by briskt** [comment]
I'm surprised ga is not defined since we should be providing the tracking ID in all IdPs. There may be something else wrong with the ga library.

**Comment by briskt** [comment]
I saw that @forevermatt marked GA as deprecated in idp-id-broker. Maybe he or @devon-sil can comment about its use here.

**Comment by forevermatt** [comment]
> I saw that @forevermatt marked GA as deprecated in idp-id-broker. Maybe he or @devon-sil can comment about its use here.

In ID Broker, I was deprecating the use of Google Analytics for reporting MFA metrics (server side, from a cron job). This all looks like client-side reporting, but I don't know if we still use that, either. 🤷

File: modules/material/themes/material/silauth/loginuserpass.twig [hunk 1]
```diff
@@ -100,9 +102,11 @@
             </span>
           </p>
 
-          <script>
-            ga('send', 'event', 'error', '{{ errorcode|e('js') }}', 'message', '{{ message|e('js') }}');
-          </script>
+          {% if analytics_tracking_id is not empty %}
```
Comment (mtompset) [inline modules/material/themes/material/silauth/loginuserpass.twig:105]: I feel like @forevermatt wanting an abstraction to make the sense of these "is not empty" clearer, even though I know what it is.

File: modules/material/themes/material/default/selectidp-links.twig [hunk 2]
```diff
@@ -15,11 +15,15 @@
 
       document.querySelector('form').appendChild(idpInput);
 
-      ga('send', 'event', 'hub', 'choice', 'IdP', id);
+      {% if analytics_tracking_id is not empty %}
+        ga('send', 'event', 'hub', 'choice', 'IdP', id);
```
Comment (hobbitronics) [inline modules/material/themes/material/default/selectidp-links.twig:19]: I think this should be 
      gtag('event', 'IdP', {
        'event_category': 'hub',
        'event_label': 'choice',
        'value': id
      });

**Review by hobbitronics — CHANGES_REQUESTED** [review]
I am finding `ga` is undefined, but in header.twig gtag is defined and has a slightly different api which looks like:
      gtag('event', 'IdP', {
        'event_category': 'hub',
        'event_label': 'choice',
        'value': id
      });

**Pull request merged**